### PR TITLE
Fixes #13605 - Configure keep alive for 443 virtual host

### PR DESCRIPTION
**Pull request by jlsherrill**

### CODE REVIEW

**Comment by jlsherrill** [comment]
I made this configurable based on the pulp's recommendation:  https://github.com/pulp/pulp/blob/master/server/etc/httpd/conf.d/pulp_apache_24.conf#L67-L79

**Comment by jlsherrill** [comment]
also going to add capsule support

File: manifests/init.pp [hunk 1]
```diff
@@ -40,6 +40,9 @@
 # $enable_ostree::      Boolean to enable ostree plugin. This requires existence of an ostree install.
 #                       type:boolean
 #
+# $max_keep_alive::     Maximum number of requests to use for the apache MaxKeepAliveRequests apache paramter
```
Comment (ehelms) [inline manifests/init.pp]: I think you can drop the second apache. ALso, a typo in "parameter"

File: templates/etc/httpd/conf.d/05-foreman-ssl.d/katello.conf.erb [hunk 2]
```diff
@@ -26,3 +26,6 @@ Alias /pub /var/www/html/pub
   ErrorDocument 500 '{"displayMessage": "Internal error, contact administrator", "errors": ["Internal error, contact administrator"], "status": "500" }'
   ErrorDocument 503 '{"displayMessage": "Service unavailable or restarting, try later", "errors": ["Service unavailable or restarting, try later"], "status": "503" }'
 </LocationMatch>
+
+KeepAlive On
+MaxKeepAliveRequests <%= @max_keep_alive %>
```
Comment (ehelms) [inline templates/etc/httpd/conf.d/05-foreman-ssl.d/katello.conf.erb]: Prefer `scope['::katello::max_keep_alive']` syntax

**Comment by jlsherrill** [comment]
@ehelms updated

File: manifests/init.pp [hunk 3]
```diff
@@ -64,6 +67,7 @@
 
   $package_names = $katello::params::package_names,
   $enable_ostree = $katello::params::enable_ostree,
+  $max_keep_alive = $katello::params::max_keep_alive,
```
Comment (ekohl) [inline manifests/init.pp]: Can be validated using ```validate_integer``` (available in stdlib >= 4.6.0)

Comment (ekohl) [inline manifests/init.pp]: Btw, not sure what kafo does or if you need type:integer as well in the documentation.

Comment (jlsherrill) [inline manifests/init.pp]: The foreman module seems to use it, but i don't see it used much in the katello module.  Went ahead and added it.

Comment (ekohl) [inline manifests/init.pp]: You need to increase the minimum version in metadata.json as well.

**Comment by beav** [comment]
This works as advertised when tested, ACK from me.

**Comment by ehelms** [comment]
Once https://github.com/Katello/puppet-katello/pull/110#discussion_r53462482 is addressed ACK

**Pull request merged**